### PR TITLE
OCPBUGS-18554: Fix "error removing HNS network when cleaning up BYOH proxy nodes"

### PR DESCRIPTION
**Pull request by saifshaikh48**
This PR re-orders a deconfiguration step removing files and HNS networks
to be after the instance has finished rebooting. This way WMCO is guaranteed
to only interact with the instance when it is reachable via SSH.

Previously, we were hitting timing issues where, after WICD cleanup is ran,
WMCO's configmap and node controllers would race, resulting in WMCO issuing
SSH commands while the node is still rebooting. These would fail and cause 
deconfiguration to reconcile again, leading to timeouts in CI for our deletion suite.

### CODE REVIEW

**Comment by saifshaikh48** [comment]
/hold 

blocked by https://github.com/openshift/windows-machine-config-operator/pull/1770

**Comment by saifshaikh48** [comment]
/unhold

https://github.com/openshift/windows-machine-config-operator/pull/1770 merged

File: pkg/windows/windows.go
Comment (sebsoto) [inline pkg/windows/windows.go:238]: This isn't really `Deconfigure` at this point. Its no longer undoing changes done by `ConfigureWICD`
It may be more apt to name it `RemoveFilesAndNetworks`

File: pkg/nodeconfig/nodeconfig.go
Comment (sebsoto) [inline pkg/nodeconfig/nodeconfig.go:534]: Theres an ordering requirement thats not clear here.
If its solely between `RunWICDCleanup` and `WaitForRebootAnnotationRemoval` it would be best to move them to a separate documented function.

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: pkg/windows/windows.go
Comment (aravindhp) [reply pkg/windows/windows.go:238]: I agree with this given that we no longer have `Windows.Configure()`. In some ways you are doing the opposite of `windows.Bootstrap()`.

File: pkg/nodeconfig/nodeconfig.go
Comment (aravindhp) [reply pkg/nodeconfig/nodeconfig.go:534]: At this point even `nc.generateWICDKubeconfig()` needs to move into the separate documented function.

**Review by aravindhp — CHANGES_REQUESTED** [review]

File: pkg/nodeconfig/nodeconfig.go
Comment (saifshaikh48) [reply pkg/nodeconfig/nodeconfig.go]: Makes sense, thanks for pointing this out

**Review by aravindhp — APPROVED** [review]

**Review by alinaryan — COMMENTED** [review]
Does this need any testing? Otherwise lgtm

**Comment by saifshaikh48** [comment]
@alinaryan the testing I did was 
- making sure the proxy job passes e2e
- running the deletion tests a bunch of times to make sure the BYOH removal flake wasn't seen anymore

**Pull request merged**
